### ISSUE TITLE
Add a marshalling example to "HttpRequest and HttpResponse / Processing responses" documentation

### ISSUE THREAD

**Issue by PiotrJander** [akka/akka-http#2842]
In the [documentation page](https://doc.akka.io/docs/akka-http/current/client-side/request-and-response.html#creating-requests) on creating HTTP requests with `akka-http` client functionality, it would be extremely  valuable if there was an example on how to include a marshalable object as a request entity. E.g.

```scala
import akka.http.scaladsl.unmarshalling.Unmarshal
import akka.http.scaladsl.marshallers.sprayjson.SprayJsonSupport._
import spray.json.DefaultJsonProtocol._
import akka.http.scaladsl.marshalling.Marshal

case class Pet(name: String)
implicit val petFormat = jsonFormat1(Pet)

for {
  entity <- Marshal[Pet("Mamon").to[RequestEntity]
  request <- HttpRequest(
    method = HttpMethods.POST,
    uri = "https://userservice.example/users",
    entity = entity
  )
} yield request
```

I have not made a pull request for this as I wouldn't be able to provide a Java version anyway.

There is a [documentation page](https://doc.akka.io/docs/akka-http/current/common/marshalling.html#using-marshallers) about using marshallers, but it would be very handy to have those two pieces of information presented together.

**Comment by jrudolph** [comment]
Yes, agreed that would be useful. We would have to find a proper place where to put it.
